src/main/java/org/reaktivity/nukleus/socks/internal/SocksController.java

```diff
@@ -107,16 +117,35 @@ public String name()
         {
             final JsonParser parser = new JsonParser();
             final JsonElement element = parser.parse(extension);
+
             if (element.isJsonObject())
             {
                 final JsonObject object = (JsonObject) element;
                 final String address = gson.fromJson(object.get("address"), String.class);
                 final int port = gson.fromJson(object.get("port"), Integer.class);
-
-                routeEx = routeExRW.wrap(extensionBuffer, 0, extensionBuffer.capacity())
-                                   .address(address)
-                                   .port(port)
-                                   .build();
+                InetAddress inet = lookupName(address);
```

Comment:
This is happening too early, it needs to be avoided when a `domain` `name` is provided instead of an IP address, otherwise it triggers a DNS lookup.

Reply:
done with nested if/else